prow/plank/controller.go

```diff
@@ -540,20 +540,12 @@ func (c *Controller) RunAfterSuccessCanRun(parent, child *kube.ProwJob, ca confi
 		// Not sure what is more correct here. Run the child for now.
 		return true
 	}
-	if ps.RunIfChanged == "" {
-		return true
-	}
-	changesFull, err := ghc.GetPullRequestChanges(org, repo, prNum)
+	shouldRun, err := ps.ShouldRun(parent.Spec.Refs.BaseRef, "", config.NewGitHubDeferredChangedFilesProvider(ghc, org, repo, prNum))
```

Comment:
This isn't technically the same as it didn't check that the job should have run against the branch in the first place or that it set `always_run: true`.

Reply:
🤢  wish we could just change this method to return false /s